docs/development.md

```diff
@@ -344,3 +344,145 @@ respectively.
 ```
+
+## Retrieving a previously-built binary and its provenance
+
+Every time a pull-request is merged to `main` the
+`oak_functions_freestanding_bin` binary is built and uploaded to `ent`.
+Similarly, a
+[signed SLSA3+ provenance is generated](https://github.com/slsa-framework/slsa-github-generator/blob/04f1fe0c7b7902c9a95f4c7eef2dc04cf0f8e6a7/internal/builders/generic/README.md)
+for it and uploaded to GitHub. The signed provenance is also uploaded to the
+public instance of Rekor hosted by Sigstore at https://rekor.sigstore.dev, and
+can be downloaded from it.
+
+This section describes how the binary and its provenance can be downloaded.
+
+Once a pull-request is merged to `main` and all CI steps are executed on branch
+`main`, an automated comment is added to the pull-request containing the hash of
+the `oak_functions_freestanding_bin` binary:
+
+```bash
+d059c38cea82047ad316a1c6c6fbd13ecf7a0abdcc375463920bd25bf5c142cc  ./oak_functions_freestanding_bin/target/x86_64-unknown-none/release/oak_functions_freestanding_bin
+```
+
+### Step 1: Download the binary
+
+The hash of the binary
+(`d059c38cea82047ad316a1c6c6fbd13ecf7a0abdcc375463920bd25bf5c142cc`) can be used
+to download the binary from `ent`, with the following command using the
+[ent HTTP API](https://github.com/google/ent#raw-http-api):
+
+```bash
+$ curl --output oak_functions_from_ent  https://ent-server-62sa4xcfia-ew.a.run.app/raw/sha256:d059c38cea82047ad316a1c6c6fbd13ecf7a0abdcc375463920bd25bf5c142cc
+  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                 Dload  Upload   Total   Spent    Left  Speed
+100 2327k    0 2327k    0     0  3637k      0 --:--:-- --:--:-- --:--:-- 3636k
+```
+
+### Step 2: Download the provenance
+
+The provenance can be downloaded from Rekor or the GitHub actions workflow that
+generated the binary and its provenance (i.e.,
+[Build Provenance](https://github.com/project-oak/oak/actions/workflows/provenance.yaml))
+
+#### Download from Rekor
+
+To fetch the provenance from Rekor, one has to first find the Rekor LogEntries
+that refer to the binary (via its SHA256 hash), using the `search` command
+provided by `rekor-cli`
+([installation instructions](https://docs.sigstore.dev/rekor/installation/)):
+
+```bash
+$ rekor-cli search --sha d059c38cea82047ad316a1c6c6fbd13ecf7a0abdcc375463920bd25bf5c142cc
+Found matching entries (listed by UUID):
+24296fb24b8ad77a872690e11780e927d9eddd1bc3a598e5490ad1c75fe039289a5dccc5b4d71576
+```
+
+Usually there is only one entry, but it is possible to get multiple entires.
+Alternatively, one can use the SHA1 Git commit hash to find the provenances
+corresponding to that commit hash:
+
+```bash
+$ rekor-cli search --sha 1b128fb2556e4bdcc4f92552654bfbca9d2fb8c6
+Found matching entries (listed by UUID):
+24296fb24b8ad77a872690e11780e927d9eddd1bc3a598e5490ad1c75fe039289a5dccc5b4d71576
+```
+
+Note that `rekor-cli` by default uses https://rekor.sigstore.dev as the Rekor
```

Comment:
Hmm... Why do I need to note this (at this point)?

Reply:
Better clarified the point in the new version.